src/app/common/project-language-chooser/project-language-chooser.component.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ProjectLanguageChooserComponent } from './project-language-chooser.component';
+import { ProjectLocale } from '../../domain/projectLocale';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { MatSelectHarness } from '@angular/material/select/testing';
+import { MatOptionHarness } from '@angular/material/core/testing';
+
+let loader: HarnessLoader;
+let component: ProjectLanguageChooserComponent;
+let fixture: ComponentFixture<ProjectLanguageChooserComponent>;
+describe('ProjectLanguageChooserComponent', () => {
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [BrowserAnimationsModule, HttpClientTestingModule, ProjectLanguageChooserComponent]
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ProjectLanguageChooserComponent);
+    loader = TestbedHarnessEnvironment.loader(fixture);
+    component = fixture.componentInstance;
+    setProjectLocale(new ProjectLocale({ default: 'en_US', supported: ['ja', 'es'] }));
+  });
+
+  it('shows available languages and selects the default language', async () => {
```

Comment:
Should this be wrapped in a describe that describes the initial state like ```describe('when the language chooser is first loaded')``` or something.